packages/dds/map/src/test/mocha/browser.spec.ts

```diff
@@ -0,0 +1,78 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+// Extend window interface to include testsLoaded property
+declare global {
+	interface Window {
+		testsLoaded?: boolean;
+	}
+}
+
+// This export makes this file a module, which is required for global augmentations
+export {};
+
+console.log("Starting browser test loader...");
+
+// Auto-discover all test files using webpack's require.context at build time
+function getTestFiles(): (() => Promise<unknown>)[] {
+	/* eslint-disable @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-return */
+	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+	// @ts-ignore - webpack specific API not in TypeScript types
+	const testContext = require.context("./", false, /\.spec\.ts$/);
+	return testContext
+		.keys()
+		.filter(
+			(file: string) =>
+				file !== "./browser.spec.ts" && // Exclude self
+				!file.includes("Utils") && // Exclude utility files
+				!file.includes("dirname"), // Exclude helper files
+		)
+		.map((file: string) => async () => testContext(file));
+	/* eslint-enable @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-return */
+}
+
+async function loadAllTests(): Promise<void> {
+	console.log("Loading all test files...");
+
+	try {
+		const testLoaders = getTestFiles();
+		console.log(`Auto-discovered ${testLoaders.length} test files`);
+
+		// Load all discovered test files
+		await Promise.all(testLoaders.map(async (loader) => loader()));
+
+		console.log("All test files loaded successfully");
+		// Signal that loading is complete
+		window.testsLoaded = true;
+		window.dispatchEvent(new CustomEvent("testsLoaded"));
+	} catch (error) {
+		// ignore error that has node:fs
+		if (error instanceof Error && error.message.includes("node:fs")) {
```

Comment:
We'll need to figure out a solution to this limitation. The Whiteboard team built out some infrastructure to polyfill `fs` for testing purposes a few years back. If there isn't a good off-the-shelf solution for this now-a-days, that might be a good place to start.